### PR TITLE
Do not introduce breaking changes in AuthenticationParamters class

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: tests/Test.ADAL.Integration.net45/HeadlessTetss/AuthenticationParamtersTest.cs [hunk 1]
```diff
@@ -45,18 +45,35 @@ public async Task AuthenticationParametersCanBeDiscovered()
...
+        {
+            string protectedApi = "https://graph.microsoft.com/v1.0/me/";
+            AuthenticationParameters ap = new AuthenticationParameters(null, null);
+
+            await ap.CreateFromResourceUrlAsync(
+                new Uri(protectedApi))
+                .ConfigureAwait(false);
 
+            // Authority might change, but should be a rare occurence
+            Assert.AreEqual(ap.Authority, "https://login.microsoftonline.com/common/oauth2/authorize");
```
Comment (jmprieur) [inline tests/Test.ADAL.Integration.net45/HeadlessTetss/AuthenticationParamtersTest.cs:72]: and here this is not an authority, it's an endpoint. See also https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/issues/1410. But that's ok. it's what the resource returns.

**Review by jmprieur — APPROVED** [review]
LGTM. Thanks for avoiding a 6.0
Still we'll have a 5.1.

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationParameters.cs [hunk 2]
```diff
@@ -55,19 +55,23 @@ public sealed class AuthenticationParameters
         /// <summary>
         /// Gets or sets the address of the authority to issue token.
         /// </summary>
-        public string Authority { get; }
+        public string Authority { get; set; }
```
Comment (bgavrilMS) [inline src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationParameters.cs:58]: Setter was previously exposed.

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationParameters.cs [hunk 3]
```diff
@@ -55,19 +55,23 @@ public sealed class AuthenticationParameters
...
         {
             ModuleInitializer.EnsureModuleInitialized();
         }
 
-        private AuthenticationParameters(string authority, string resource)
+        /// <summary>
+        /// Ctor
+        /// </summary>
+        [Obsolete("Please use the static builders of this class, not the constructor", false)]
+        public AuthenticationParameters(string authority, string resource)
```
Comment (bgavrilMS) [inline src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationParameters.cs:74]: A ctor was not present in 5.0 :)

**Review by jennyf19 — APPROVED** [review]
Will you merge it later into dev?

**Comment by bgavrilMS** [comment]
Yes, I will merge to dev.

**Pull request merged**